lib/phobos/actions/process_message.rb

```diff
@@ -57,10 +57,17 @@ def force_encoding(value)
       def process_message(payload)
         instrument('listener.process_message', @metadata) do
           handler = @listener.handler_class.new
-          preprocessed_payload = handler.before_consume(payload)
+          preprocessed_payload = handler.before_consume(payload, @metadata)
+          consume_block = Proc.new { handler.consume(preprocessed_payload, @metadata) }
 
-          @listener.handler_class.around_consume(preprocessed_payload, @metadata) do
-            handler.consume(preprocessed_payload, @metadata)
+          if @listener.handler_class.respond_to?(:around_consume)
+            # around_consume class method implementation
+            warn "#{Kernel.caller.first} around_consume has been moved to instance method, please update your consumer."\
```

Comment:
I think we can follow the same pattern as in PhobosDBCheckpoint here, namely define a deprecate method in Phobos module

```
    def deprecate(message)
      warn "DEPRECATION WARNING: #{message} #{Kernel.caller.first}"
    end
```

Then we can call it using `Phobos.deprecate("around_consume has been moved to instance method, please update your consumer " \ .....)`

It is easier to have one single point of doing deprecations and prefixing with DEPRECATION WARNING catches one's eye more reliably.

Reply:
Done!